langchain/llms/azureml_endpoint.py

```diff
@@ -0,0 +1,142 @@
+
+        url = self.endpoint_url
+        # Replace this with the primary/secondary key or AMLToken for the endpoint
+        api_key = self.endpoint_api_key
+        if not api_key:
+            raise Exception("A key should be provided to invoke the endpoint")
+
+        # The azureml-model-deployment header will force the request to go to a specific deployment.
+        # Remove this header to have the request observe the endpoint traffic rules
+
+        headers = {'Content-Type': 'application/json', 'Authorization': ('Bearer ' + api_key), 'azureml-model-deployment': self.deployment_name}
+        
+        req = urllib.request.Request(url, body, headers)
+        try:
+            response = urllib.request.urlopen(req, timeout=50)
+            result = response.read()
+        except urllib.error.HTTPError as error:
+            print("The request failed with status code: " + str(error.code))
+            result = str(error)
+        except Exception as e:
+            print("Calling Azure Managed Online endpoint failed!")
+            result = str(e)
+        return result
+
+
+class AzureMLModel(LLM, BaseModel):
+    """Wrapper around Azure ML Hosted models using Managed Online Endpoints.
+
+    Example:
+        .. code-block:: python
+
+            auzre_llm = AzureMLModel(
+                endpoint_url="https://<your-endpoint>.<your_region>.inference.ml.azure.com/score",
+                endpoint_api_key="my-api-key",
+                deployment_name="my-deployment-name",
+                catalog_type="my-catalog-type")
+    """
+
+    endpoint_url: str = None
+    """ URL of prexisting Endpoint """
+    
+    endpoint_api_key: str = None
+    """ Authentication Key for Endpoint"""
+    
+    deployment_name: str = None
+    """ Deployment Name for Endpoint"""
+
+    catalog_type: str = None
+    """ Model Catalog Type: hugging_face or open_source """
+
+    http_client: Any = None  #: :meta private:
+    
+    model_kwargs: Optional[dict] = None
+    """Key word arguments to pass to the model."""
+
+    @validator("http_client", always=True, allow_reuse=True)
+    @classmethod
+    def validate_client(cls, field_value, values) -> Dict:
+        """Validate that api key and python package exists in environment."""
+        endpoint_key = get_from_dict_or_env(
+            values, "endpoint_api_key", "ENDPOINT_API_KEY"
+        )
+        endpoint_url = values["endpoint_url"]
+        deployment_name = values["deployment_name"]
+        http_client = AzureMLEndpointClient(endpoint_url, endpoint_key, deployment_name)
+        return http_client
+
+    def format_body(self, prompt, parameters) -> Dict:
+        if self.catalog_type == "open_source":
+            return {"inputs": {"input_string": [prompt]}, "parameters": parameters} 
+        elif self.catalog_type == "hugging_face":
+            options = {"use_cache": parameters["use_cache"] or True, "wait_for_model": parameters["wait_for_model"] or False}
```

Comment:
This might error if use_cache/wait_for_model are not present, and using or could change the desired values (i.e. if parameters["use_cache"] is False, then use_cache in options will still be true due to the or with True)